modules/parquet/src/parquetjs/reader.ts

```diff
@@ -406,10 +408,42 @@ export class ParquetEnvelopeReader {
     ) as any;
 
     const pagesOffset = Number(colChunk.meta_data?.data_page_offset!);
-    const pagesSize = Number(colChunk.meta_data?.total_compressed_size!);
-    const pagesBuf = await this.read(pagesOffset, pagesSize);
+    let pagesSize = Number(colChunk.meta_data?.total_compressed_size!);
 
-    return await decodeDataPages(pagesBuf, field, compression);
+    if (!colChunk.file_path) {
+      pagesSize = Math.min(
+        this.fileSize - pagesOffset,
+        Number(colChunk.meta_data?.total_compressed_size)
+      );
+    }
+
+    const opts = {
+      type,
+      rLevelMax: field.rLevelMax,
+      dLevelMax: field.dLevelMax,
+      compression,
+      column: field,
+      num_values: colChunk.meta_data?.num_values,
+      dictionary: null
+    };
+
+    let dictionary;
+
+    if (colChunk?.meta_data?.dictionary_page_offset) {
```

Comment:
If you understand this code we are bringing in maybe add a one line comment explaining what it does?

```// Look for dictionary in the page```

Reply:
Add comment and refactored this part